moto/dynamodb2/models.py

```diff
@@ -110,27 +111,27 @@ def describe_attrs(self, attributes):
         }
 
     def update(self, update_expression, expression_attribute_names, expression_attribute_values):
-        ACTION_VALUES = ['SET', 'set', 'REMOVE', 'remove']
-
-        action = None
-        for value in update_expression.split():
-            if value in ACTION_VALUES:
-                # An action
-                action = value
-                continue
-            else:
+        parts = [p for p in re.split(r'\b(SET|REMOVE|ADD|DELETE)\b', update_expression) if p]
+        for action, valstr in zip(parts[:-1:2], parts[1::2]):
```

Comment:
`zip` will not raise an error if the iterables are not of the same length: 

```python
zip(range(10), range(20)) # [(0, 0) ... (9, 9)]
```

Perhaps it would be better to use a list comprehension?

```python
for operation, attributes in ((parts[index], parts[index + 1]) for index in xrange(len(parts))):
    ...
```